hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/pipeline/ClosePipelineSubcommand.java

```diff
@@ -51,7 +51,7 @@ public class ClosePipelineSubcommand extends ScmSubcommand {
   public void execute(ScmClient scmClient) throws IOException {
     if (!Strings.isNullOrEmpty(closeOption.pipelineId)) {
       if (filterOptions.getReplicationFilter().isPresent()) {
-        System.out.println("When close pipeline by ID filters are ignored.");
+        System.err.println("Replication filters can only be used with --all");
```

Comment:
Sorry if the last comment was not clear. We should also fail the command here instead of closing the pipeline anyways since the user did something invalid. We can probably throw `IllegalArgumentException` like `FilterPipelineOptions#getReplicationFilter` does.

Reply:
I just tried all three of these to double check and option 3 is what we want. There is logic farther up to print just the message to stderr and not the stack trace. Option 2 will print the message twice. Option 1 will exit 0 when it should exit nonzero.

Just looking at the example you provided I think cert.InfoSubcommand might print the message twice as well, but I did not try it out.